modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxManager.java

```diff
@@ -1526,7 +1526,7 @@ public void commitTx(IgniteInternalTx tx) throws IgniteCheckedException {
             removeObsolete(tx);
 
             // 7. Assign transaction number at the end of transaction.
-            tx.endVersion(cctx.versions().next(tx.topologyVersion()));
+            tx.endVersion(cctx.versions().next(tx.topologyVersion().topologyVersion()));
```

Comment:
Can be replaced with
tx.endVersion(cctx.cacheContext(tx.txState().firstCacheId()).cache().nextVersion());

Reply:
Fixed.